source/appModules/teamtalk5classic.py

```diff
@@ -0,0 +1,245 @@
+		speech.speakObject(self)
+
+	def _get_name(self):
+		return self.parent.info.name(self.IAccessibleChildID)
+
+
+class AppModule(appModuleHandler.AppModule):
+	def event_NVDAObject_init(self, obj):
+		# For some TeamTalk versions, the incoming chat control does not
+		# return the correct _isWindowUnicode flag.
+		if obj.windowClassName == "RichEdit20A":
+			obj._isWindowUnicode = False
+		# Label channel chat in/out controls.
+		if obj.windowControlID == ID_CHATOUT:
+			# Translators: a label for a read only edit control
+			# in the TeamTalk voice conferencing client where incoming chat
+			# messages are displayed.
+			obj.name = _("Channel messages")
+		elif obj.windowControlID == ID_CHATIN:
+			# Translators: a label for the TeamTalk chat entry field
+			obj.name = _("Type a message to the channel")
+		# Better name Add/Remove buttons where they occur.
+		# Example: Banned Users dialog from the Server menu.
+		elif obj.role == controlTypes.ROLE_BUTTON:
+			if obj.name == ">":
+				# Translators: A TeamTalk button label.
+				obj.name = _("Add to unban list")
+			elif obj.name == "<":
+				# Translators: A TeamTalk button label.
+				obj.name = _("Remove from unban list")
+		# Name the lists in the Banned Users dialog.
+		elif obj.role == controlTypes.ROLE_LIST:
+			if obj.windowControlID == ID_BANNEDLIST:
+				# Translators: Label for a list of banned users in TeamTalk
+				obj.name = _("Banned")
+			elif obj.windowControlID == ID_UNBANNEDLIST:
+				# Translators: Label for a list of unbanned users in TeamTalk
+				obj.name = _("Unbanned")
+
+	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
+		# There is a VU meter progress bar in the main window
+		# that shouldn't be monitored (it presents many spurious updates).
+		if obj.windowClassName == "msctls_progress32" and (
+			obj.name == "VU"
+			or obj.windowControlID == ID_VU
+		):
+			try:
+				clsList.remove(ProgressBar)
+			except ValueError:
+				from logHandler import log
+				log.exception("Couldn't remove VU meter progress bar")
+		# Name toolbar icons.
+		elif obj.windowControlID == ID_TOOLBAR:
+			# Applies to toolbar and its items.
+			if obj.role == controlTypes.ROLE_TOOLBAR:
+				clsList.insert(0, TTToolBar)
+			elif obj.parent and obj.parent.role == controlTypes.ROLE_TOOLBAR:
+				clsList.insert(0, TTToolBarItem)
+
+	@scriptHandler.script(
+		gesture="kb:control+shift+s",
+		description="""Says various on/off states from the toolbar. Depending
+on TeamTalk version, these can include connection status,
+status of voice activation, push-to-talk, and video
+features, whether channel audio is being saved to files,
+and whether the channel message window is showing.
+On one press, just says which are checked (in effect).
+On two presses, says all states.
+		"""
+	)
+	def script_sayToolbarInfo(self, gesture):
+		info = TTToolBarInfo()
+		if not info:
+			return
+		scnt = scriptHandler.getLastScriptRepeatCount()
+		if scnt > 0:
+			[
+				speech.speakObject(o)
```

Comment:
I have the feeling that a list comprehension isn't suitable for this. This will result in a list of Nones. Please change this into a real for loop.